test/test.hpp

```diff
@@ -93,7 +93,13 @@ inline void expect_op(const TLeft& left,
 #define EXPECT_EQUAL(LEFT, RIGHT) EXPECT_OP(LEFT, ==, RIGHT)
 #define STATUS(...) EXPECT((__VA_ARGS__) == 0)
 
+// Conflicts with GoogleTest FAIL()
+#ifdef FAIL
+#undef FAIL
```

Comment:
How it could be so that test/test.hpp is used in gtest?

Reply:
Perhaps we can refactor these defs out into their own header, so that they are only included when they are needed

Reply:
> Perhaps we can refactor these defs out into their own header, so that they are only included when they are needed

@JehandadKhan separated Macros defined in test.hpp to a separate header test_macro.hpp and updated affected code.